buildrunner/docker/runner.py

```diff
@@ -251,6 +254,94 @@ def cleanup(self):
 
         self.container = None
 
+    def restore_caches(self, caches: OrderedDict, cache_archive_ext: str):  # pylint: disable=too-many-branches,too-many-locals
+        """
+        Restores caches from the host system to the destination location in the docker container.
+        """
+        if caches is None or not isinstance(caches, OrderedDict):
+            raise TypeError(f"Caches should be type OrderDict instead of {type(caches)}")
+
+        restored_cache_src = set()
+        for local_cache_archive_file, docker_path in caches.items():
+            if docker_path in restored_cache_src:
+                print(f"Destination path {docker_path} has already been matched and restored to the container "
+                      f"skipping {local_cache_archive_file} -> {docker_path}")
+                continue
+
+            # Check for prefix matching
+            if not os.path.exists(local_cache_archive_file):
+                cache_dir = os.path.dirname(local_cache_archive_file)
+
+                if not os.path.exists(cache_dir):
+                    print(f"Cache directory {cache_dir} does not exist, "
+                          f"skipping restore of archive {local_cache_archive_file}")
+                    continue
+
+                files = [f for f in listdir(cache_dir) if isfile(join(cache_dir, f))]
+
+                cache_key = local_cache_archive_file\
+                    .replace(f"{cache_dir}/", "")\
+                    .replace(f".{cache_archive_ext}", "")
```

Comment:
This can be replaced with:

```
# at the top of the file
from pathlib import Path

# in the code
cache_key = Path(local_cache_archive_file).stem
```

Reply:
- [ ] refactor cache key assignment

Reply:
That is cool I didn't know about that property